app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -248,7 +250,7 @@ class SettingsFragment : PreferenceFragmentCompat() {
                 SettingsFragmentDirections.actionSettingsFragmentToSearchEngineFragment()
             }
             resources.getString(R.string.pref_key_tracking_protection_settings) -> {
-                requireContext().metrics.track(Event.TrackingProtectionSettings)
+                TrackingProtection.etpSettings.record(NoExtras())
```

Comment:
Not sure replacing the current telemetry with recording Glean events in place is the right approach.
If we are not using anymore `metrics.track` we are sidestepping the `metrics` configuration

https://github.com/mozilla-mobile/fenix/blob/7ca7190199a9b5ce980b73f4cf3a3c15ac65f3d0/app/src/main/java/org/mozilla/fenix/components/Analytics.kt#L102-L112

 for a [wrapper with multiple responsibilities](https://github.com/mozilla-mobile/fenix/blob/7ca7190199a9b5ce980b73f4cf3a3c15ac65f3d0/app/src/main/java/org/mozilla/fenix/components/metrics/MetricController.kt#L41): 
 - registering multiple analytics services for tracking the same events.
 - checking whether the telemetry should be tracked - users can disable it from app settings.
 - allowing to also easily eanble telemetry depending on build configuration.
 
 --- 
One important piece of Glean functionality that I'd want to also confirm is what happens with a `record` call if Glean is not initialized/doesn't have upload enabled because maybe users disabled the telemetry in app settings. Would Glean just cache the telemetry for this period and then send it when telemetry is enabled again? 
@travis79 Do you know?

Reply:
- We do not send any events directly to Adjust and if we do not plan to in the future IMO always trying to send through both services is inefficient so, on that topic, I do not think we need the controller.

- Glean is already initialized with the value of the `isTelemetryEnabled` so checking it in the controller is redundant. Glean will not send the events if `uploadEnabled`  is false. Maybe @travis79 can clarify what happens with the metrics at that point. 

There might be a problem with changing the setting after Glean is initialized. I do not think we currently reset Glean at that point (@travis79  is calling `setUploadEnabled(false)` recommended  or is it expected that the change is made next time Glean is initialized?) 

- Enabling depending on the build configuration can be as easily configured through `uploadEnabled` parameter on Glean init. 


That being said, I would prefer to try to remove all the wrapping code on the calls. But at the same time, we could keep the controller structure and implement different `track` methods that accept  `EventMetricType<T, E>` as params. 

Maybe @travis79 can clarify some of the issues stated here. (Thanks in advance! )

Reply:
If Glean is not initialized, it will cache API calls and play them back once it is initialized (provided telemetry collection is enabled). This has some limits, but nothing we have run into problems with on mobile.

You are correct that trying to wrap recording of metrics in `should we record` conditionals is redundant, the correct way to control this is through the `initialize` function, and through the `setUploadEnabled` API in Glean.
Glean takes care of all of the logic of what the correct thing to do with the metrics is, depending on the state of the preference. When `uploadEnabled` is false, no metrics are recorded or sent. If it is toggled to false during runtime, then Glean will stop recording metrics, delete any metrics it already has recorded, and send a "deletion-request" ping to the server to indicate we should delete all the user's data there also. Calling `setUploadEnabled` during runtime is expected and correct, it should be directly wired to the "Send Usage Data" preference in the app settings.

The Glean team has long advocated for calling the Glean API functions directly at the site of measurement rather than having a wrapper around things to make diagnosing telemetry issues easier and to prevent relying on Glean internals, but we also understood that at the time the wrapper was created we were also sending some duplication of data to Adjust, LeanPlum and Glean and having a Tee was useful.  I defer to the product team to make the best decision here for their purposes, though.

Reply:
`Glean.initialized` _must_ be called on startup every time, regardless of the state of the preference in order to ensure that Glean is doing the correct things to follow our data-collection policies such as sending the "deletion-request" ping.